beets/dbcore/query.py

```diff
@@ -447,6 +447,24 @@ def match(self, item):
         return any([q.match(item) for q in self.subqueries])
 
 
+class NotQuery(MutableCollectionQuery):
```

Comment:
As hinted on the main comment, the reason for being a subclass of `MutableCollectionQuery` is rather weak, and mainly due to laziness. Conceptually, I can't really think of an argument for `NotQuery` having more than one subquery (and if there is a case for that, one could always use a `NotQuery(AndQuery([...]))` or similar).

The "*only one subquery*" is not enforced at the moment, and has some impact on readability (`self.clause_with_joiner(joiner)` never uses the joiner, `self.match()` iterates over only one element, constructor receives a list of one element as its first parameter, etc). I'm still revising the implications, but seems like preserving the rest of niceties and special methods of `MutableCollectionQuery` (or at least `CollectionQuery`) would be desirable. Any hints on what would be the best option? May just be a case of overriding `__init__`, accepting only one subquery.